src/cmatrix.cpp

```diff
@@ -62,417 +62,939 @@
+    //--------------------------- Use libtiff to read input file-----------------------------
+    //---------------------------------------------------------------------------------------
+    else{
+        cout<<" The input data is processed using libtiff library"<<endl;
+        unsigned int h,w,x=0,y=0;
+        unsigned int tileWidth, tileLength;
+        unsigned short int spp=0,bps=0;
+        TIFF *tif = NULL;
+        unsigned char *buf8, *buf8tiled;
+        unsigned short *buf16, *buf16tiled;
+        RGBcolor rgb = {0,0,0};
+        ImageMatrix R_matrix, G_matrix, B_matrix;
+        Moments2 R_stats, G_stats, B_stats;
+
+        TIFFSetWarningHandler(NULL);
+        if( (tif = TIFFOpen(filename, "r")) ) {
+            source = filename;
+
+            TIFFGetField(tif, TIFFTAG_IMAGEWIDTH, &w);
+            width = w;
+            TIFFGetField(tif, TIFFTAG_IMAGELENGTH, &h);
+            height = h;
+            TIFFGetField(tif, TIFFTAG_BITSPERSAMPLE, &bps);
+            bits=bps;
+
+            TIFFGetField(tif, TIFFTAG_TILEWIDTH, &tileWidth);
+            TIFFGetField(tif, TIFFTAG_TILELENGTH, &tileLength);
+
+            if ( ! (bits == 8 || bits == 16) ) {
+                cout<<"Only bit values equal to 8 and 16 are supported"<<endl;
+                return (0); // only 8 and 16-bit images supported.
+            }
+            TIFFGetField(tif, TIFFTAG_SAMPLESPERPIXEL, &spp);
+            if (!spp) spp=1;  /* assume one sample per pixel if nothing is specified */
+            // regardless of how the image comes in, the stored mode is HSV
+            if (spp == 3) {
+                ColorMode = cmHSV;
+                // If the bits are > 8, we do the read into doubles so that later
+                // we can scale the image to its actual signal range.
+                if (bits > 8) {
+                    R_matrix.ColorMode = cmGRAY;
+                    R_matrix.allocate (width, height);
+                    G_matrix.ColorMode = cmGRAY;
+                    G_matrix.allocate (width, height);
+                    B_matrix.ColorMode = cmGRAY;
+                    B_matrix.allocate (width, height);
+                }
+            } else {
+                ColorMode = cmGRAY;
+            }
+            if ( TIFFNumberOfDirectories(tif) > 1) return(0);   /* get the number of slices (Zs) */
+
+            /* allocate the data */
+            allocate (width, height);
+            writeablePixels pix_plane = WriteablePixels();
+            writeableColors clr_plane = WriteableColors();
+
+            if (tileWidth < width && tileLength < height){
+                if (tileWidth != 1024) cout<<" Warning: tileWidth is not 1024 and instead is="<< tileWidth<<endl;
+                if (tileLength != 1024) cout<<" Warning: tileLength is not 1024 and instead is="<< tileLength<<endl;
+
+                cout<<" It is tiled tiff format processed by libtiff library"<<endl;
```

Comment:
Can you guard your print outs with DEBUG?

Reply:
Per our conversation, I will leave this for now